### PR TITLE
Update DG

### LINKED ISSUE #54: Update DG
Add uml diagrams to DG and explain.
Add implementation details of commands.

### PR DESCRIPTION
**Pull request by Benson15912**
Fix #54 
Updated our DG to fit requirements:
https://nus-cs2103-ay2324s2.github.io/website/admin/tp-deliverables.html#deliverable-developer-guide-dg

Added:
- Acknowledgement of my code reuse.
- Implementation of certain features: Add Volunteer feature, Find Volunteer Feature, Assign volunteer feature
- Removed proposed features: undo/redo
- Added use cases
- Added instructions for manual testing
- Added  Appendix: Effort

However, planned enhancements are left empty as of now.

### CODE REVIEW

File: docs/images/AssignActivityDiagram.png
Comment (SimKianSeng) [inline docs/images/AssignActivityDiagram.png:1]: This one is it possible to have a single red box `Shows error message`?

**Review by SimKianSeng — COMMENTED** [review]
Minor comment, otherwise LGTM 👍🏻

**Review by SimKianSeng — APPROVED** [review]

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -323,57 +310,287 @@ Priority: High (must have) - `***`, Medium (nice to have) - `**`, Low (unlikely
...
+2. User enters the person to remove.
+3. Match asks for confirmation.
+4. User confirms deletion.
+5. The system displays a success message and removed the volunteer from the app.
+
+Use Case ends.
+
+**Extensions**:
+
+- 2a. User enters an invalid volunteer.
```
Comment (YowSiaoKang) [inline docs/DeveloperGuide.md:393]: I think the updated version will process invalid volunteer after confirmation. Will need to update.

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -403,39 +627,247 @@ testers are expected to do more *exploratory* testing.
...
-1. Saving window preferences
+2. Saving window preferences
 
    1. Resize the window to an optimum size. Move the window to a different location. Close the window.
 
    1. Re-launch the app by double-clicking the jar file.<br>
        Expected: The most recent window size and location is retained.
 
-1. _{ more test cases …​ }_
+3. Exit Application
```
Comment (YowSiaoKang) [inline docs/DeveloperGuide.md:641]: should we use the `exit` CLI command instead of the GUI

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -403,39 +627,247 @@ testers are expected to do more *exploratory* testing.
...
+   6. Test Case: `assign d/ a/[Existing availability]`
+
+        Expected: Error thrown, stating that details cannot be empty.
+
+### View list of assignments
+1. Views the list of every assignment.
+   1. Test Case: `lista`
+
+        Expected: The list of all assignments is shown on the GUI.
+   2. Test Case: `lista [extra inputs`
```
Comment (YowSiaoKang) [inline docs/DeveloperGuide.md:780]: missing close bracket for `[extra inputs]`

**Review by YowSiaoKang — CHANGES_REQUESTED** [review]
Just need to clarify some points

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (Benson15912) [reply docs/DeveloperGuide.md:636]: Yup changed ut

File: docs/images/AssignActivityDiagram.png
Comment (Benson15912) [reply docs/images/AssignActivityDiagram.png:1]: I tried using the labels in the puml docs, but the diagram was very buggy

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -323,57 +310,282 @@ Priority: High (must have) - `***`, Medium (nice to have) - `**`, Low (unlikely
...
+Use Case ends.
+
+**Extensions**:
+
+- 3a. User cancels deletion
+  - 3a1. The system acknowledges the cancellation and does nothing.
+  - Use case ends.
+
+- 5a. User enters an invalid volunteer.
+    - 5a1. Match shows an error message
```
Comment (YowSiaoKang) [inline docs/DeveloperGuide.md:397]: The number is wrong

Comment (Benson15912) [reply docs/DeveloperGuide.md]: Changed to 4

**Review by YowSiaoKang — APPROVED** [review]

**Pull request merged**